apps/ospTutorial/ospTutorial.cpp

```diff
@@ -552,17 +590,34 @@ void _loadPBF(Server &server)
   }
 
   const auto &featureresult = queryresult.featureresult();
+  if (!featureresult.has_spatialreference()) {
+    std::cerr
+        << "No spatial reference found - we don't know how to reproject. Ignoring"
+        << std::endl;
+    return;
+  }
+
+  const auto sr = featureresult.spatialreference();
+  if (sr.wkid() != 102100 && sr.lastestwkid() != 3857) {
+    std::cerr << "Can only handle WebMercator - aborting now." << std::endl;
+    return;
+  }
+
   std::vector<vec3f> centers;
 
+  const bool hasZ = featureresult.hasz();
   for (const auto &feature : featureresult.features()) {
     if (feature.has_geometry()) {
       if (feature.geometry().geometrytype()
           == esriPBuffer::
               FeatureCollectionPBuffer_GeometryType_esriGeometryTypePoint) {
         const auto &coords = feature.geometry().coords();
-        const auto origin = server.updateOrigin(coords[0], coords[1], 0);
-        const auto position = vec3d(coords[0], coords[1], 1);
-        centers.emplace_back(position - origin);
+        auto position = vec3d(coords[0], coords[1], (hasZ ? coords[2] : 1));
+        convert::webMercatorToSphericalECEF(position);
+        const auto origin = server.updateOrigin(position);
+        centers.emplace_back(position.x - origin.x,
+            position.y - origin.y,
+            (hasZ ? position.z - origin.z : 1));
```

Comment:
It would really be much nicer, if the spheres would also show up for `z>=0`. I don't like these extra if statements.